File: DemoUnitTest/DemoUnitTest/DemoUnitTest/ViewController2.m
//
//  ViewController2.m
//  DemoUnitTest
//
//  Created by zj－db0465 on 15/11/25.
//  Copyright © 2015年 icetime17. All rights reserved.
//

#import "ViewController2.h"

@interface ViewController2 ()

@end

@implementation ViewController2

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor lightGrayColor];
    
    _btnBack = [[UIButton alloc] initWithFrame:CGRectMake(0, self.view.frame.size.height - 100, self.view.frame.size.width, 50)];
    [_btnBack setTitle:@"Go to ViewController" forState:UIControlStateNormal];
    [_btnBack setTitleColor:[UIColor blueColor] forState:UIControlStateNormal];
    [_btnBack setTitleColor:[UIColor redColor] forState:UIControlStateHighlighted];
    [_btnBack addTarget:self action:@selector(actionBack:) forControlEvents:UIControlEventTouchUpInside];
    _btnBack.layer.borderColor = [UIColor redColor].CGColor;
    _btnBack.layer.borderWidth = 2.0f;
    [self.view addSubview:_btnBack];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)actionBack:(UIButton *)sender {
    [self dismissViewControllerAnimated:NO completion:nil];
}

@end

File: DemoSwift/DemoSwift/DemoSwift/ViewController2.swift
//
//  ViewController2.swift
//  DemoSwift
//
//  Created by zj－db0465 on 15/11/17.
//  Copyright © 2015年 icetime17. All rights reserved.
//

import UIKit

typealias MyIntroduce = (String, Int) -> Void

class ViewController2: UIViewController {
    
    var myIntroduce: MyIntroduce!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.backgroundColor = UIColor.whiteColor()
        
        // Do any additional setup after loading the view.
        
        let btn = UIButton(frame: CGRectMake(0, self.view.frame.size.height - 100, self.view.frame.size.width, 50))
        btn.setTitle("Go back to ViewController", forState: UIControlState.Normal)
        btn.setTitleColor(UIColor.blueColor(), forState: UIControlState.Normal)
        btn.setTitleColor(UIColor.redColor(), forState: UIControlState.Highlighted)
        btn.layer.borderColor = UIColor.redColor().CGColor
        btn.layer.borderWidth = 2.0
        btn.addTarget(self, action: Selector("actionButton:"), forControlEvents: UIControlEvents.TouchUpInside)
        self.view.addSubview(btn)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func actionButton(sender: UIButton) {
        self.dismissViewControllerAnimated(true) { () -> Void in
            self.myIntroduce("ViewController2", 2)
        }
    }

}
